core/src/functions/content.functions.ts

```diff
@@ -909,60 +946,112 @@ export const generateContextRanges = (
                         sectionLevel
-                    ].completeStartingRange = generateRangeFromGenericSection(
-                        context[sectionLevel].startingSections[0]
+                    ]!.completeStartingRange = generateRangeFromGenericSection(
+                        context[sectionLevel]!.startingSections[0]!
                     );
                 }
                 if (
-                    context[sectionLevel].startingSections[
-                        context[sectionLevel].startingSections.length - 1
-                    ].phraseEndId > lastPhraseId
+                    context[sectionLevel]!.startingSections[
+                        context[sectionLevel]!.startingSections.length - 1
+                    ]!.phraseEndId > lastPhraseId
                 ) {
                     contextRanges.sections[
                         sectionLevel
-                    ].completeEndingRange = generateRangeFromGenericSection(
-                        context[sectionLevel].startingSections[
-                            context[sectionLevel].startingSections.length - 1
-                        ]
+                    ]!.completeEndingRange = generateRangeFromGenericSection(
+                        context[sectionLevel]!.startingSections[
+                            context[sectionLevel]!.startingSections.length - 1
+                        ]!
                     );
                 }
             }
 
-            if (context[sectionLevel] && context[sectionLevel].previousSections.length)
+            if (context[sectionLevel]! && context[sectionLevel]!.previousSections.length)
                 contextRanges.sections[
                     sectionLevel
-                ].previousRange = generateRangeFromGenericSection(
-                    context[sectionLevel].previousSections[
-                        context[sectionLevel].previousSections.length - 1
-                    ]
+                ]!.previousRange = generateRangeFromGenericSection(
+                    context[sectionLevel]!.previousSections[
+                        context[sectionLevel]!.previousSections.length - 1
+                    ]!
                 );
-            if (context[sectionLevel] && context[sectionLevel].nextSections.length)
-                contextRanges.sections[sectionLevel].nextRange = generateRangeFromGenericSection(
-                    context[sectionLevel].nextSections[0]
+            if (context[sectionLevel]! && context[sectionLevel]!.nextSections.length)
+                contextRanges.sections[sectionLevel]!.nextRange = generateRangeFromGenericSection(
+                    context[sectionLevel]!.nextSections[0]!
                 );
         }
     }
     return contextRanges;
 };
 
+/**
+ * normalizes document contents to only use necessary properties (similar to
+ * `stripUnnecessaryDataFromBibleContent`) and to only use the value "after" in `phrase.skipSpace`
+ * @param {DocumentElement[]} contents
+ * @returns {DocumentElement[]}
```

Comment:
agreed to: remove jsdoc comments as long as the typescript annotates it, since they can get out of date, etc